File: YegnaBet.Domain/Entities/Location.cs
namespace YegnaBet.Domain.Entities
{
    public class Location
    {
        public long Id { get; set; }
        public string Country { get; set; } = "ethiopia";
        public string City { get; set; } = null!;
        public string Area { get; set; } = null!;
        public string? SubArea { get; set; }
    }
}

File: yegnabet-web/src/types/common/location.ts
/**
 * @description an interface used to fetch location information to/from
 *  a backend service. A location is but a string of unique names for every listing
 *  that exists and avoid duplication when handling listings with similar locations.
 */
export interface ListingLocation {
    id: number;
    country: string;
    city: string;
    area: string;
    subArea?: string;
    count: number;
}